eZ/Publish/Core/Repository/User/PasswordHasher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\User;
+
+use eZ\Publish\API\Repository\Values\User\User;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+
+/**
+ * @internal
+ */
+final class PasswordHasher implements PasswordHasherInterface
```

Comment:
The logic here is pretty simple but I'm missing unit tests (or maybe is covered somewhere else?)  😉